### PR TITLE
Adding tests for Agent options and validation.

### PR DESCRIPTION
**Pull request by cheftako**

### CODE REVIEW

File: cmd/agent/app/options/options_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
+package options
```
Comment (andrewsykim) [inline cmd/agent/app/options/options_test.go:1]: Missing license header here.

I think `cmd/server/app/options/options_test.go` might be missing license header too.

Comment (cheftako) [reply cmd/agent/app/options/options_test.go:17]: Sadly more than just these two. Did a quick scan and added license headers where I found them to be missing.

**Comment by tallclair** [comment]
We might want to copy https://github.com/kubernetes/kubernetes/blob/master/hack/verify-boilerplate.sh into this repo.

File: cmd/agent/app/options/options_test.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import (
+	"testing"
+	"reflect"
+	"time"
+	"github.com/stretchr/testify/assert"
+	"fmt"
+)
+
+func TestDefaultServerOptions(t *testing.T) {
+	defaultAgentOptions := NewGrpcProxyAgentOptions()
```
[the remark this answers is not in the document]
Comment (cheftako) [reply cmd/agent/app/options/options_test.go:32]: Done

**Pull request merged**